zkevm-circuits/src/circuit_tools/cell_manager.rs

```diff
@@ -70,67 +107,389 @@ impl<F: Field> Expr<F> for &Cell<F> {
     }
 }
 
-/// CellType
-#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub enum CellType {
-    /// General
-    Storage,
+#[derive(Clone, Debug)]
+pub struct CellConfig<C: CellTypeTrait> {
+    cell_type: C,
+    num_columns: usize,
+    phase: u8,
+    is_permute: bool,
+}
+
+impl<C: CellTypeTrait> CellConfig<C> {
+    pub fn init_columns<F: Field>(&self, meta: &mut ConstraintSystem<F>) -> Vec<Column<Advice>>{
+        let mut columns = Vec::with_capacity(self.num_columns);
+        match self.phase {
+            0 => columns.iter_mut()
+                    .map(|c| *c = meta.advice_column_in(FirstPhase))
+                    .collect::<Vec<()>>(),
+            1 => columns.iter_mut()
+                    .map(|c| *c = meta.advice_column_in(SecondPhase))
+                    .collect::<Vec<()>>(),
+            2 => columns.iter_mut()
+                    .map(|c| *c = meta.advice_column_in(ThirdPhase))
+                    .collect::<Vec<()>>(),
+            _ => unreachable!(),
+        };
```

Comment:
Mapping to the phase first, and then simply setting that phase for all columns I think will simplify this code.

Reply:
You mean group together all columns in the same phase? 
I'm trying to group them by CellType so that they're easy to find in query_cells API.

Reply:
I mean something like this:
```
let phase = match self.phase {
  0 => FirstPhase,
  1 => SecondPhase,
  2 => ThirdPhase,
  _ => unreachable!(),
};
let columns = vec![meta.advice_column_in(phase); self.num_columns];
```

No idea if this compiles, but you get the idea.